clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -264,6 +264,71 @@ def ConstPtrAttr : CIR_Attr<"ConstPtr", "ptr", [TypedAttrInterface]> {
   let hasCustomAssemblyFormat = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// ThreeWayCmpInfoAttr
+//===----------------------------------------------------------------------===//
+
+def ThreeWayCmpOrderingKind_Strong : I32EnumAttrCase<"Strong", 1, "strong">;
+def ThreeWayCmpOrderingKind_Partial : I32EnumAttrCase<"Partial", 2, "partial">;
+
+def ThreeWayCmpOrderingKind : I32EnumAttr<
+  "ThreeWayCmpOrderingKind", "three-way comparison ordering kind",
+  [ThreeWayCmpOrderingKind_Strong, ThreeWayCmpOrderingKind_Partial]
+> {
+  let cppNamespace = "::mlir::cir";
+}
+
+def ThreeWayCmpInfoAttr
```

Comment:
This is nice! It's also possible we could do this inside the Op definition, example:

Instead of
```
%{{.+}} = cir.three_way_cmp(%{{.+}}, %{{.+}}, #cir.three_way_cmp_info<strong, #cir.int<-1> : !s8i, #cir.int<0> : !s8i, #cir.int<1> : !s8i>) : !s32i, !s8i
```

It's more readable to do something like:

```
%{{.+}} = cir.three_way_cmp(%{{.+}} : !s32i, %{{.+}}, strong, lt = -1, eq = 0, gt = 1) : !s8i
or // if using the attribute
%{{.+}} = cir.three_way_cmp(%{{.+}} : !s32i, %{{.+}}, #cir.cmp3way_info) : !s8i
// This could be achieved by adding an attribute alias 
```

I like the attribute because then all cmp3way in a TU will likely share the same one, so we don't need to create these new values all the time.

- For `strong`, `lt, eq, gt`: (a) use record printing and (b) is there a real need for the values to be CIR int typed already? If not, for simplicity, seems like we could keep them as MLIR integers attributes up to LoweringPrepare stage.
- Note that I also moved the operand type inline.

Reply:
> Use record printing

Updated.

> seems like we could keep them as MLIR integers attributes up to LoweringPrepare stage

I tried using `mlir::IntegerAttr` but the syntax is still a little verbose. `mlir::IntegerAttr` will be printed as something like `42 : si8` instead of just `42`. So instead I use `int64_t` directly.

> Note that I also moved the operand type inline.

Updated.